src/secondary.rs

```diff
@@ -759,6 +794,401 @@ impl<'a, K: Key, V: 'a + Copy> Extend<(K, &'a V)> for SecondaryMap<K, V> {
+    pub fn get(&self) -> &V {
+        unsafe { self.map.get_unchecked(self.key.clone()) }
+    }
+
+    /// Gets a mutable reference to the value in the entry.
+    ///
+    /// If you need a reference to the OccupiedEntry which may outlive the destruction of the Entry value, see into_mut.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use slotmap::*;
+    /// # use slotmap::secondary::Entry;
+    ///
+    /// let mut sm = SlotMap::new();
+    /// let mut sec = SecondaryMap::new();
+    ///
+    /// let k = sm.insert(1);
+    /// sec.insert(k, 10);
+    ///
+    /// if let Entry::Occupied(mut o) = sec.entry(k).unwrap() {
+    ///     *o.get_mut() = 20;
+    ///     assert_eq!(o.get(), &20);
+    /// }
+    /// ```
+    pub fn get_mut(&mut self) -> &mut V {
+        unsafe { self.map.get_unchecked_mut(self.key.clone()) }
+    }
+
+    /// Converts the OccupiedEntry into a mutable reference to the value in the entry with a lifetime bound to the map itself.
+    ///
+    /// If you need multiple references to the OccupiedEntry, see get_mut.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use slotmap::*;
+    /// # use slotmap::secondary::Entry;
+    ///
+    /// let mut sm = SlotMap::new();
+    /// let mut sec = SecondaryMap::new();
+    ///
+    /// let k = sm.insert(2);
+    /// sec.insert(k, 0);
+    ///
+    /// let mut v = &mut 3;
+    /// if let Entry::Occupied(o) = sec.entry(k).unwrap() {
+    ///     v = o.into_mut(); // v outlives the entry
+    /// }
+    ///
+    /// *v = 1;
+    ///
+    /// assert_eq!(sec[k], 1);
+    /// ```
+    pub fn into_mut(self) -> &'a mut V {
+        unsafe { self.map.get_unchecked_mut(self.key) }
+    }
+
+    /// Sets the value of the entry, and returns the entry's old value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use slotmap::*;
+    /// # use slotmap::secondary::Entry;
+    ///
+    /// let mut sm = SlotMap::new();
+    /// let mut sec = SecondaryMap::new();
+    ///
+    /// let k = sm.insert(1);
+    /// sec.insert(k, 10);
+    ///
+    /// if let Entry::Occupied(mut o) = sec.entry(k).unwrap() {
+    ///     let v = o.insert(20);
+    ///     assert_eq!(v, 10);
+    ///     assert_eq!(o.get(), &20);
+    /// }
+    /// ```
+    pub fn insert(&mut self, mut value: V) -> V {
+        let old_value = self.get_mut();
+        std::mem::swap(&mut value, old_value);
+        value
+    }
+
+    /// Takes the value out of the entry, and returns it.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use slotmap::*;
+    /// # use slotmap::secondary::Entry;
+    ///
+    /// let mut sm = SlotMap::new();
+    /// let mut sec = SecondaryMap::new();
+    ///
+    /// let k = sm.insert(1);
+    /// sec.insert(k, 10);
+    ///
+    /// if let Entry::Occupied(mut o) = sec.entry(k).unwrap() {
+    ///     let v = o.remove();
+    ///     assert_eq!(v, 10);
+    ///     assert_eq!(sec.get(k), None);
+    /// }
+    /// ```
+    pub fn remove(self) -> V {
```

Comment:
I feel this goes against the spirit of `Entry` as `self.map.remove` checks whether the slot exists, whether it has the right version, etc. I think you should be able to directly do:

    let keyd = key.into();
    let slot = self.map.slots.get_unchecked_mut(keyd.idx as usize);
    slot.version -= 1;
    self.map.num_elems -= 1;
    std::mem::replace(&mut slot.value, None).unwrap()